src/downloader/sentry_client_reactor.rs

```diff
@@ -191,56 +191,120 @@ impl Drop for SentryClientReactor {
+            EventLoopStreamId::Receive,
+            Box::pin(receive_stream.map_ok(EventLoopStreamResult::Receive)),
+        );
+        stream.insert(
+            EventLoopStreamId::Stop,
+            Box::pin(stop_stream.map(|result| Ok(EventLoopStreamResult::Stop(result)))),
+        );
+
+        while let Some((stream_id, result)) = stream.next().await {
+            match stream_id {
+                EventLoopStreamId::Send => {
+                    // process an outgoing message that has been just sent
+                    match result {
+                        Ok(EventLoopStreamResult::Send(sent_peers_count)) => {
+                            debug!(
+                                "SentryClientReactor.EventLoop sent message to {:?} peers",
+                                sent_peers_count
+                            );
                         }
+                        Ok(_) => panic!("unexpected result {:?}", result),
                         Err(error) => {
-                            error!("SentryClientReactor.EventLoop sentry.send_message error: {}", error);
+                            error!(
+                                "SentryClientReactor.EventLoop sentry.send_message error: {}",
+                                error
+                            );
                         }
                     }
                 }
-                message_result = in_stream.next() => {
-                    match message_result {
-                        Some(Ok(message_from_peer)) => {
+                EventLoopStreamId::Receive => {
+                    // process an incoming message that was received
+                    match result {
+                        Ok(EventLoopStreamResult::Receive(message_from_peer)) => {
                             let id = message_from_peer.message.eth_id();
                             debug!("SentryClientReactor.EventLoop incoming message: {:?}", id);
 
-                            if self.receive_messages_senders
-                                .read()
-                                .get(&id)
-                                .ok_or_else(|| anyhow::anyhow!("SentryClientReactor.EventLoop unexpected message id {:?}", id))?
-                                .send(message_from_peer.message).is_err() {
+                            let receive_messages_senders = self.receive_messages_senders.read();
+                            let sender_opt = receive_messages_senders.get(&id);
+                            if sender_opt.is_none() {
+                                anyhow::bail!(
+                                    "SentryClientReactor.EventLoop unexpected message id {:?}",
+                                    id
+                                );
+                            }
+                            let sender = sender_opt.unwrap();
+
+                            let send_sub_result = sender.send(message_from_peer.message);
+                            if send_sub_result.is_err() {
```

Comment:
```suggestion
                            if sender.send(message_from_peer.message).is_err() {
```

Reply:
It reads more naturally, but if you feel that having intermediate variable is for the better, let's run with it.